### PR TITLE
Creates a menu item to export the wallets transfers to CSV

### PR DESCRIPTION
**Pull request by darcys22**
Utilises the export_transfers RPC wallet method that returns a string
which is simply saved to a file prompted to the user.

### CODE REVIEW

File: src/components/menus/wallet_settings.vue [hunk 1]
```diff
@@ -647,6 +727,32 @@ export default {
...
+      this.hideModal("export_transfers");
+
+      let passwordDialog = await this.showPasswordConfirmation({
+        title: this.$t("dialog.exportTransfers.title"),
+        noPasswordMessage: this.$t("dialog.exportTransfers.message"),
+        ok: {
+          label: "export".toLocaleUpperCase(this.locale),
+          color: "primary"
+        },
+        dark: this.theme == "dark",
```
Comment (kylezs) [inline src/components/menus/wallet_settings.vue:740]: Screen Shot 2021-04-27 at 9 18 36 am

This should be white, with the same blue button... ik a few of the other ones I missed are still dark, but yeah, should be white

File: src/components/menus/wallet_settings.vue [hunk 2]
```diff
@@ -647,6 +727,32 @@ export default {
...
+        color: this.theme == "dark" ? "white" : "dark"
+      });
+      passwordDialog
+        .onOk(password => {
+          // if no password set
+          password = password || "";
+          if (this.modals.key_image.type == "Export")
```
Comment (kylezs) [inline src/components/menus/wallet_settings.vue:747]: Is this not `modals.export_transfers` ?

Comment (kylezs) [reply src/components/menus/wallet_settings.vue:747]: Export transfers isn't working for me on Mac. When I click "Confirm" the popup just disappears as if it was dismissed and nothing happens. I'm gueessing the problem is in this area of the code somewhere.

Comment (kylezs) [reply src/components/menus/wallet_settings.vue:747]: > Is this not `modals.export_transfers` ?
In fact, why is this `if` statement necessary at all? There is no toggle when exporting the transfers

Comment (darcys22) [reply src/components/menus/wallet_settings.vue:747]: Yeah this if statement shouldnt be there. From the export key images modal which has an "Export" and "Import" radio button.

File: src/components/menus/wallet_settings.vue (hunk 1 above)
Comment (darcys22) [reply src/components/menus/wallet_settings.vue:740]: Confirming both buttons should be white if dark mode is selected?

Comment (kylezs) [reply src/components/menus/wallet_settings.vue:740]: The button colours can stay the same tbh, just the background to be dark. "show private keys" is an example of what I mean.

File: src-electron/main-process/modules/wallet-rpc.js [hunk 3]
```diff
@@ -2470,6 +2473,79 @@ export class WalletRPC {
...
+      (err, password_hash) => {
+        if (err) {
+          this.sendGateway("show_notification", {
+            type: "negative",
+            i18n: "notification.errors.internalError",
+            timeout: 2000
+          });
+          return;
+        }
+        if (!this.isValidPasswordHash(password_hash)) {
```
Comment (kylezs) [inline src-electron/main-process/modules/wallet-rpc.js:2492]: `else if` otherwise there could be several errors shown on top of each other.

**Review by kylezs — APPROVED** [review]

**Pull request merged**